mmtk/src/api.rs

```diff
@@ -58,9 +58,9 @@ pub extern "C" fn alloc(mutator: *mut Mutator<SelectedPlan<JikesRVM>>, size: usi
 }
 
 #[no_mangle]
-pub extern "C" fn post_alloc(mutator: *mut Mutator<SelectedPlan<JikesRVM>>, refer: ObjectReference, type_refer: ObjectReference,
+pub extern "C" fn post_alloc(mutator: *mut Mutator<SelectedPlan<JikesRVM>>, refer: ObjectReference, _type_refer: ObjectReference,
```

Comment:
Is there any specific reason for marking `type_refer` as unused argument, instead of removing it?

Reply:
It depends on what API JikesRVM expects, and what MMTk provides. I think it is reasonable that JikesRVM expects a `type_refer` parameter in `post_alloc()` the same as Java MMTk (https://github.com/mmtk/jikesrvm/blob/06a52fa2d26635187d2ca4cd93012b33ce90bb4a/rvm/src/org/jikesrvm/runtime/SysCall.java#L282, vs, https://github.com/mmtk/jikesrvm/blob/06a52fa2d26635187d2ca4cd93012b33ce90bb4a/MMTk/src/org/mmtk/plan/MutatorContext.java#L216). But mmtk-core does not need that paramter in its `post_alloc()`. Thus, in the binding, we dont use `type_refer`. 

I am happy to make the change through, and that would need changes in our JikesRVM fork.

Reply:
With your explanation, I think we should keep the argument as unused, because ultimately we want to make minimal changes to the JikesRVM source code to simplify upstreaming

Reply:
Correct. But the `post_alloc()`/`sysPostAlloc()` API is what we added. Basically that means we can modify it. I was just trying to keep it the same as the original `postAlloc()` in Java MMTk.